### PR TITLE
smb: fix wrong endian conversion when parse NTLM Negotiate Flags - V4

### PR DESCRIPTION
**Pull request by b1tg**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to redmine ticket: [5783](https://redmine.openinfosecfoundation.org/issues/5783)

Previous PR: https://github.com/OISF/suricata/pull/8449

Changes from last PR:
- fix commit message
- update the user guide

Signed-off-by: Lancer Cheng b1tg@protonmail.ch

suricata-verify-pr: 1105
https://github.com/OISF/suricata-verify/pull/1105

### CODE REVIEW

**Comment by jufajardini** [comment]
Updated PR description to the accepted format for SV PR ;)

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -1210,37 +1210,31 @@ NTLMSSP fields
 * "domain" (string): the Windows domain.
 * "user" (string): the user.
 * "host" (string): the host.
+* "version" (string): the version of client.
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:1213]: nit: replace "the version of client" with "the client version"

File: doc/userguide/output/eve/eve-json-format.rst [hunk 2]
```diff
@@ -1210,37 +1210,31 @@ NTLMSSP fields
...
-      "user": "administrator",
-      "host": "BLU"
-    }
-
+  "ntlmssp": {
+      "domain": "CONTOSO",
+      "user": "Administrator",
+      "host": "SERVER01",
+      "version": "6.0 build 6001 rev 15"
+  }
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:1222]: This CI error here: https://github.com/OISF/suricata/actions/runs/4073611297/jobs/7023535861#step:14:276

Is probably due to:
not having a new line between `} ` and `More complete example`.

File: doc/userguide/output/eve/eve-json-format.rst [hunk 3]
```diff
@@ -1210,37 +1210,31 @@ NTLMSSP fields
...
-      "user": "administrator",
-      "host": "BLU"
-    },
-    "request": {
-      "native_os": "Unix",
-      "native_lm": "Samba 3.9.0-SVN-build-11572"
-    },
-    "response": {
-      "native_os": "Windows (TM) Code Name \"Longhorn\" Ultimate 5231",
-      "native_lm": "Windows (TM) Code Name \"Longhorn\" Ultimate 6.0"
```
Comment (jufajardini) [inline doc/userguide/output/eve/eve-json-format.rst:1243]: I would prefer that we didn't remove this part of the example unless there's a good reason for it, as having it makes the example more complete.
Can we just adjust the example to contain the new version info?

**Review by jufajardini — CHANGES_REQUESTED** [review]
Thanks for incorporating the requested changes :)

I have left a comment pointing out what I think is causing the CI failure. If you want to build the documentation locally to make sure it's fine before the next PR, you'll need sphinx-build (get and install it, then re-run `./configure`, and then you can run `make html`, for instance, to check that documentation is built alright.

Apart from that, one other inline comment regarding the json example.

**Comment by b1tg** [comment]
move to https://github.com/OISF/suricata/pull/8516

File: doc/userguide/output/eve/eve-json-format.rst (hunk 3 above)
Comment (b1tg) [reply doc/userguide/output/eve/eve-json-format.rst:1243]: check https://github.com/OISF/suricata/pull/8516

**Pull request closed without merging**